Feature/flippable wiki cards

spergel opened the pull request:
- Implemented 3D card flip animation where clicking on card outside of key areas brings up iframe with wikipedia article
- Removed close button from back interface

## Future considerations
- Mobile-optimized Wikipedia URLs (needs additional media query handling)
- Performance optimizations for low-powered devices
- Accessibility enhancements for screen readers

## Testing notes
Tested on:
- Desktop Chrome/Firefox/Safari
- Mobile Safari/Chrome
- Various screen sizes (needs mobile URL handling in future)" \
  --web

IsaacGemal commented:
Oh this is great.
It will def kill my vercel bill, but I think it's worth it.

spergel commented:
Hmm Let me think if there is a better way to do this without it being as intensive on vercel

IsaacGemal commented:
Also for the mobile just do a device size check with tailwind and replace the url with the mobile version

IsaacGemal commented:
Forget vercel, it looks really bad on mobile. Let me first figure out the mobile stuff first.

spergel commented:
I think we can just have it take up like 5/6 instead of 2/3 of the page and use mobile webpages instead of what currently exists.

spergel commented:
Okay I added something which ~should~ be better. I widened the screen on mobile, moved the share button down to the bottom of the page, and added a back button because i feel like theres not too much room to click back otherwise. 

Possible Alterations:
 - buttons on the top of the page instead of the bottom
 - have mobile wikipedia take up entire width of screen instead of 9/10
 - find some other way to display wikipedia on mobile

IsaacGemal commented:
I'll have to think about this a bit. Probably would be best if it's like optional (you can click the card or read the wiki page)

Solarin-Johnson commented:
Yo! I'd love to help refine the UI with pure CSS. Should I give it a shot? :)

IsaacGemal commented:
I've thought about it and the flipping is a bit excessive especially on really large displays, I might put this on hold.
Doesn't flow very well with the scrolling idea.
